extensions/src/main/java/org/pentaho/platform/web/servlet/UploadFileUtils.java

```diff
@@ -586,6 +588,32 @@ private long getFolderSize( File folder ) {
     return foldersize;
   }
 
+  /**
+   * <p>Removes the file name, leaving all extensions of a given filename.</p>
+   * <p>Returns <code>null</code> if filename is <code>null</code>, and an empty string if the filename is empty or has
+   * no extensions.</p>
+   *
+   * @param filename the filename for which we want the extensions
+   * @return all extensions of the given filename
+   */
+  protected String removeFileName( String filename ) {
+    if ( null != filename ) {
+      int dotIndex = filename.indexOf( DOT );
```

Comment:
Shouldn't we be looking for the `lastIndexOf` the dot instead?

Reply:
To anwser your question above: if the filename is `luis.martins.2019.jpg`, the extension we want to get is `.martins.2019.jpg`, that's why I named the method "remove file name".

We need to keep the extensions because of the way `tar` files are handled (they rely on the extensions and not on some magic number/signature on the content itself like other supported file types).

Theoretically, we'd only need the last two extensions, however the cost of obtaining a maximum of two would be much greater than getting all of them. That's why I'm using `indexOf` and not `lastIndexOf`.

Reply:
OK, we're trying to accommodate for `.tar.gz` files?
What if my file is `luis.martins.2019.tar.gz`? Will that work?

Reply:
The compressed files extensions supported here are: `.tar`, `.zip`, `.tgz`, `.gz` and `.gzip`.
Although it wouldn't be difficult to support `.tar.gz`, you'd have to change it to `.tgz`. 

So... if you'd asked about `luis.martins.2019.tgz` the answer would be Yes and No :-)

- **Yes**, the file would be correctly uploaded and the temp files would be created (named something like "_username-918273645._**martins.2019.tgz**_.tmp_" for the original file and "_username-192837465._**martins.2019**_.tmp_" for the "contained"/"_targazed_" file).

- **No**, as the "_targazed_" file has `.2019` extension, an error would occur in this particular process (as expected) as it's not in the supported ones (`.csv`, `.dat` and `.txt`).